### PR TITLE
Scheduler implementation of Continual Pre-Training of Large Language Models: How to (re)warm your model? 

### PR DESCRIPTION
**Pull request by jinwonkim93**
Scheduler implementation of Continual Pre-Training of Large Language Models: How to (re)warm your model? (https://arxiv.org/pdf/2308.04014.pdf)

# Description
almost identical to consine min lr but it adds constant ratio which freezes lr at some percent of training step.

## Motivation and Context
This scheduler has been proven for continual pretraining.

## How has this been tested?
Made test code

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -797,6 +797,7 @@ early_stopping_patience: 3
 lr_scheduler: # 'one_cycle' | 'log_sweep' | empty for cosine
 lr_scheduler_kwargs:
 cosine_min_lr_ratio: # decay lr to some percentage of the peak lr, e.g. cosine_min_lr_ratio=0.1 for 10% of peak lr
+cosine_constant_lr_ratio: # freeze lr at some percentage of the step, e.g. cosine_constant_lr_ratio=0.8 means start cosine_min_lr at 80% of training step
```
Comment (winglian) [inline README.md]: Would be nice to have a link to the arxiv paper referenced here too

**Review by winglian — COMMENTED** [review]
Thanks! Lgtm

File: README.md (hunk 1 above)
Comment (jinwonkim93) [reply README.md]: i put paper link to it.

**Review by winglian — COMMENTED** [review]
There is a test failure w 

E       TypeError: AxolotlTrainingArguments.__init__() got an unexpected keyword argument 'cosine_constant_lr_ratio'

**Comment by jinwonkim93** [comment]
> There is a test failure w
> 
> E TypeError: AxolotlTrainingArguments.**init**() got an unexpected keyword argument 'cosine_constant_lr_ratio'

thank you i add in arguments

**Review by winglian — APPROVED** [review]

**Pull request merged**